DataStructures/Queues/Deque.java

```diff
@@ -0,0 +1,163 @@
+/**Implementation of double ended queue*/
+
+import java.util.Scanner;
+
+public class Deque {
+	private int maxSize;
+	private int f, r, a[];
+
+	public Deque(int S) {
+		maxSize = S;
+		a = new int[S];
+		f = r = -1;
+	}
+
+	int deleteRear() {
+		if (f == -1)
+			return -1;
+		else if (f == r) {
+			int x = a[r];
+			f = r = -1;
+			return x;
+		} else {
+			int x = a[r];
+			r--;
+			return x;
+		}
+	}
+
+	int insertFront(int x) {
+		if (f == 0)
+			return 0;
+		else if (f == -1 || r == -1) {
+			f++;
+			r++;
+			a[f] = x;
+			return x;
+		} else {
+			f--;
+			a[f] = x;
+			return x;
+		}
+	}
+
+	int insertRear(int x) {
+		if (r == maxSize - 1)
+			return -1;
+		else if (f == -1 || r == -1) {
+			f++;
+			r++;
+			a[r] = x;
+			return x;
+		} else {
+			r++;
+			a[r] = x;
+			return x;
+
+		}
+	}
+
+	int deleteFront() {
+		if (f == -1)
+			return -1;
+		else if (f == r) {
+			int x = a[f];
+			f = r = -1;
+			return x;
+		} else {
+			int x = a[f];
+			f++;
+			return x;
+		}
+	}
+
+	int display() {
+		if (f == -1 && r == -1)
+			return -1;
+		else {
+			for (int i = f; i <= r; i++)
+				System.out.println(a[i]);
+			return 0;
+		}
+
+	}
+
+	public static void main(String[] args) {
```

Comment:
Instead of main function , you should have written test cases , which will be far better than using main function